src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -603,18 +603,15 @@ protected static bool Matches(ISymbol? symbol1, ISymbol notNullOriginalUnreduced
         var semanticFacts = state.SemanticFacts;
         var semanticModel = state.SemanticModel;
 
-        return IsInNamespaceOrTypeContext()
-            ? SymbolUsageInfo.Create(GetTypeOrNamespaceUsageInfo())
-            : GetSymbolUsageInfoCommon();
+        var topNameNode = node;
+        while (syntaxFacts.IsQualifiedName(topNameNode.Parent))
+            topNameNode = topNameNode.Parent;
 
-        bool IsInNamespaceOrTypeContext()
```

Comment:
inlined this method as we want it's 'topmost parent' node result for another op we do.